Add instructions on compiling with dynamic lib + fix floating point error

Atreyagaurav opened the pull request:
Fixes: #123 
Fixes: #142 

Tested with these changes that are not in the commit (due to the unavailability of the static library):

```diff
modified   CMakeLists.txt
@@ -46,9 +46,6 @@ if (UNIX)
         set(fdebug          "-Wall")
         set(frelease        "-O")
         set(FFC             "gnu")
-        if(NOT APPLE)
-            link_libraries("-static")
-        endif()
     endif()
 elseif(WIN32)
     if(CMAKE_Fortran_COMPILER_ID STREQUAL Intel)
```

The test runs completely now (no floating point error), but the test fails due to the values being different. I only tested it in `gfortran` so please verify this does not break anything for `ifx` compiler.

The log is attached: [LastTest.log](https://github.com/user-attachments/files/24155208/LastTest.log)

Atreyagaurav commented:
Note: I think the failed test is probably related to #144 and not to the changes in this pull request.

EDIT: We could probably test multiple commits (going back) to narrow where the output changes occur. I tried to do that for 61.0.1, but I ran into more floating errors, that are probably fixed now, so I did not want to go over fixing them to see if I get the same error.

From my test, it is at least before the commit: a2682b3, I ran into compilation errors while trying to compile some commits before that (and the git tree looks really convoluted), so I did not continue.

odav commented:
> Fixes: #123 Fixes: #142
> 
> Tested with these changes that are not in the commit (due to the unavailability of the static library):
> 
> ```diff
> modified   CMakeLists.txt
> @@ -46,9 +46,6 @@ if (UNIX)
>          set(fdebug          "-Wall")
>          set(frelease        "-O")
>          set(FFC             "gnu")
> -        if(NOT APPLE)
> -            link_libraries("-static")
> -        endif()
>      endif()
>  elseif(WIN32)
>      if(CMAKE_Fortran_COMPILER_ID STREQUAL Intel)
> ```
> 
> The test runs completely now (no floating point error), but the test fails due to the values being different. I only tested it in `gfortran` so please verify this does not break anything for `ifx` compiler.
> 
> The log is attached: [LastTest.log](https://github.com/user-attachments/files/24155208/LastTest.log)

There will be a new PR to be more OS specific on static library creation, having flags for  Arch, Debian, RH, etc. to take care of gnu OS variants and library dependencies. This will make it obsolete to modify `CMakelists.txt`

Atreyagaurav commented:
Hi @fgeter , thank you for the reviews. 

I use Arch Linux, I can test it when the new solution is pushed, no problem. 

I have already modified the code to use exp_w from utils, please refer to the git diff tab. I rebased it to keep the git commits clean.

fgeter commented:
Atreyagaurav

One of the issues is the swatplus main repo releases are ancient and badly
out of date.  Olaf talked to Jeff Arnold yesterday about the need to cut a
new release.  I think that is going to happen soon.   It might be good to
have a meeting on the testing issue.  We have talked internally about
automating some tests in GitHub that would have to pass before a release is
built.  I'm not sure what github CI entails.  Olaf is the github expert and
we would need to engage him on that.

Thanks
Frank

On Wed, Jan 7, 2026 at 2:58 PM Zero ***@***.***> wrote:

> *Atreyagaurav* left a comment (swat-model/swatplus#143)
> <https://github.com/swat-model/swatplus/pull/143#issuecomment-3720976990>
>
> Thank you!
>
> For the tests, you are right, we can not test all distributions. Covering
> the main ones, and leaving the instructions for them, should help others
> fix the issue with their distribution. That's why I wanted to add the
> static flag in the documentation, as I couldn't find that detail while
> building. I thought I needed to recompile gcc tools for that (not fun)
>
> While talking about testing, I was thinking that we could add github CI to
> the repo. This is also related to #144
> <https://github.com/swat-model/swatplus/issues/144>, I couldn't figure
> out which commit corresponds to the changes that breaks the ctest from
> the history. So having CI should help with that. That way we could test at
> least windows/mac and ubuntu. Not sure if we can test multiple distro that
> way. But just making sure it compiles and runs the test should be a good
> plus.
>
> I can work on the github CI once #144
> <https://github.com/swat-model/swatplus/issues/144> is resolved. The CI
> itself should be simple, but I can not work on it now because I don't know
> if the test is actually failing because of error or the results need to be
> updated (because the estimates are better now). The first option seems to
> be the likely case.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/swat-model/swatplus/pull/143#issuecomment-3720976990>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A2J6UF2UMTOHDFEN2LBCBZT4FV6OXAVCNFSM6AAAAACPASNEKOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTOMRQHE3TMOJZGA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Thanks
Frank Geter
